example/src/App.tsx

```diff
@@ -0,0 +1,65 @@
+/* @refresh granular */
+import { createSignal, onCleanup, onMount } from "solid-js";
+import { CounterProvider, useCounter } from "./CounterContext";
+
+const title = 'Count';
+
+function Count() {
+  const counter = useCounter();
+  onMount(() => {
+    console.log('Mounted Count');
+  });
+  onCleanup(() => {
+    console.log('Unmounted Count');
+  });
+  return (
+    <h1>{title}: {counter.value()}</h1>
+  );
+}
+
+function Increment() {
+  const counter = useCounter();
+  onMount(() => {
+    console.log('Mounted Increment');
+  });
+  onCleanup(() => {
+    console.log('Unmounted Increment');
+  });
+  return (
+    <button onClick={counter.increment}>
+      Increment
+    </button>
+  );
+}
+
+function Decrement() {
+  const counter = useCounter();
+  onMount(() => {
+    console.log('Mounted Decrement');
+  });
+  onCleanup(() => {
+    console.log('Unmounted Decrement');
+  });
+  return (
+    <button onClick={counter.decrement}>
+      Decrement
+    </button>
+  );
+}
+
+export default function App() {
```

Comment:
This minimal example is to test the following:
- Context lookup is working in HMR
- Granular mode test